compare_platforms/compare_platforms_read_only.c

```diff
@@ -0,0 +1,202 @@
+    strcpy(newpublicreview, "5 ***** strong accept. This author deserves two Nobels and an ice cream.\n");
+
+#ifdef __CHERI_PURE_CAPABILITY__
+    printf("\nsmallsz=%zx, CRRL(smallsz)=%zx\n", smallsz,
+        __builtin_cheri_round_representable_length(smallsz));
+    printf("biggersz=%zx, CRRL(biggersz)=%zx\n", biggersz,
+        __builtin_cheri_round_representable_length(biggersz));
+    printf("username=%#p realname=%#p diff=%tx\n", username, realname, realname - username);
+    printf("realname=%#p privatereview=%#p diff=%tx\n", realname, privatereview, privatereview - realname);
+    printf("privatereview=%#p publicreview=%#p diff=%tx\n", privatereview, publicreview, publicreview - privatereview);
+    printf("publicreview=%#p permissions=%#p diff=%tx\n", publicreview, permissions, permissions - publicreview);
+    printf("realname=%#p permissions=%#p diff=%tx\n", realname, permissions, permissions - realname);
+    
+    if(can_write(&review) == false) {
+        printf("\nSetting the review to read-only.\n");
+        struct review *ro_review = set_read_only(&review);
+        assert((cheri_perms_get(ro_review) & CHERI_PERM_STORE) == 0);
+
+        //printf("\nTry anyway? y/n: \n");
+        //char answer;
+        //scanf(" %c", &answer);
+        //if (answer == 'Y' || answer == 'y'){
+        if ((argc > 1) && (strcmp(argv[1], "-overwrite") == 0)) {
+            printf("\nThe struct is read-only so trying to change the review will make the program crash.\n");
+            bWeak = false;
+            b_improved = change_publicreview(ro_review, newpublicreview, bWeak);
+            // This line should be unreachable.
+            print_details(ro_review);
+        }
+        fflush(stdout);
+        b_improved = false;
+    }
+#else
+    printf("\nusername=%p realname=%p diff=%tx\n", username, realname, realname - username);
+    printf("realname=%p privatereview=%p diff=%tx\n", realname, privatereview, privatereview - realname);
+    printf("privatereview=%p publicreview=%p diff=%tx\n", privatereview, publicreview, publicreview - privatereview);
+    printf("publicreview=%p permissions=%p diff=%tx\n", publicreview, permissions, permissions - publicreview);
+    printf("realname=%p permissions=%p diff=%tx\n", realname, permissions, permissions - realname);
+    
+    b_improved = change_publicreview(&review, newpublicreview, true);
+#endif 
+
+    // Contrast the behaviour of this code in a CHERI vs. non-CHERI environment.
+    if(b_improved == false) {
+        overflow_reviewer_realname(smallsz+1, 1, &review);
+    
+        const size_t oversz = review.permissions - review.realname + 2 - smallsz;
+        overflow_reviewer_realname(smallsz+2, oversz, &review);
+        
+        // If we reached this line, we should have acquired write privileges on the review.
+        b_improved = change_publicreview(&review, newpublicreview, bWeak);
+        print_details(&review);
+    }
+    
+    printf("\nFreeing username.\n");
```

Comment:
I personally think there's too much `printf`ing going on (not only here, but throughout). I don't know if it's worth keeping so much in the merged version. @ltratt thoughts?

Reply:
Fixed in e0adbc70332ea555fd0f0fbdf355c7b7f897f79b